IoTuring/Entity/Deployments/Disk/Disk.py

```diff
@@ -103,6 +121,68 @@ def Update(self) -> None:
             attributeKey=EXTRA_KEY_DISK_FREE,
             attributeValue=usage.free,
             valueFormatterOptions=VALUEFORMATOPTIONS_DISK_GB)
+        
+        if OsD.IsLinux():
+            # because of this reverse parsing to get the devicename from the partition name, all the following is only linux having the partition named nvme* or sd* 
+            if self.configuredIo:
+                devname = self.disk_partition.device.split("/", 2)
+                if "nvme" in self.disk_partition.device:
+                    disk_io = psutil.disk_io_counters(perdisk=True)[devname[2][:-2]]
+                elif "sd" in self.disk_partition.device:
+                    disk_io = psutil.disk_io_counters(perdisk=True)[devname[2][:-1]]
+
+                self.SetEntitySensorExtraAttribute(
+                    sensorDataKey=KEY_USED_PERCENTAGE,
+                    attributeKey=EXTRA_KEY_DISK_READ_COUNT,
+                    attributeValue=disk_io.read_count)
+                
+                self.SetEntitySensorExtraAttribute(
+                    sensorDataKey=KEY_USED_PERCENTAGE,
+                    attributeKey=EXTRA_KEY_DISK_WRITE_COUNT,
+                    attributeValue=disk_io.write_count)
+
+                self.SetEntitySensorExtraAttribute(
+                    sensorDataKey=KEY_USED_PERCENTAGE,
+                    attributeKey=EXTRA_KEY_DISK_READ_BYTES,
+                    attributeValue=disk_io.read_bytes,
+                    valueFormatterOptions=VALUEFORMATOPTIONS_DISK_B)
+                
+                self.SetEntitySensorExtraAttribute(
+                    sensorDataKey=KEY_USED_PERCENTAGE,
+                    attributeKey=EXTRA_KEY_DISK_WRITE_BYTES,
+                    attributeValue=disk_io.write_bytes,
+                    valueFormatterOptions=VALUEFORMATOPTIONS_DISK_B)
+                
+                self.SetEntitySensorExtraAttribute(
+                    sensorDataKey=KEY_USED_PERCENTAGE,
+                    attributeKey=EXTRA_KEY_DISK_READ_TIME,
+                    attributeValue=disk_io.read_time,
+                    valueFormatterOptions=VALUEFORMATOPTIONS_TIME)
+                
+                self.SetEntitySensorExtraAttribute(
+                    sensorDataKey=KEY_USED_PERCENTAGE,
+                    attributeKey=EXTRA_KEY_DISK_WRTIE_TIME,
+                    attributeValue=disk_io.write_time,
+                    valueFormatterOptions=VALUEFORMATOPTIONS_TIME)
+                
+                # the following are only supported in linux
+                if OsD.IsLinux():
```

Comment:
You are already in an `if Linux` loop, I think this should be on a different tab level? Somewhere you messed up levels.

Reply:
You are correct i failed to see the first ```if linux```